src/instruction/instruction.c

```
@@ -168,11 +193,13 @@ static bool does_substate_need_to_be_displayed(substate_t *substate, public_key_
             return true;
         case SUBSTATE_TYPE_PREPARED_UNSTAKE:
             return true;
-        case SUBSTATE_TYPE_STAKE_SHARE:
+        case SUBSTATE_TYPE_STAKE_OWNERSHIP:
+        case SUBSTATE_TYPE_VALIDATOR_ALLOW_DELEGATION_FLAG:
+        case SUBSTATE_TYPE_VALIDATOR_OWNER_COPY:
             return false;
+        default:
+            return false;  // should never happen
```

Can we exit() with error rather than failing silently in these cases?